sensor/tests/resource/pod/pod_test.go

```diff
@@ -123,3 +126,70 @@ func (s *PodHierarchySuite) Test_ParentlessPodsAreTreatedAsDeployments() {
 			"Should have received four different pod events (3 from nginx-deployment and 1 from nginx-rouge")
 	})
 }
+
+func (s *PodHierarchySuite) Test_DeleteDeployment() {
+	s.testContext.RunBare("Removing a deployment should send empty violation", func(t *testing.T, testC *resource.TestContext, _ map[string]k8s.Object) {
+		deleteDep, err := testC.ApplyFileNoObject(context.Background(), resource.DefaultNamespace, NginxDeployment)
+		require.NoError(t, err)
+
+		var id string
+		// Check the deployment is processed
+		testC.LastDeploymentState("nginx-deployment", func(deployment *storage.Deployment, _ central.ResourceAction) error {
```

Comment:
This makes think that we should have an assertion for "the deployment event with name X was received", so you don't have to do it like that.

Reply:
I've added a new function `WaitForDeploymentEvent` that basically does that. Let me know what you think.